tools/crm_attribute.c

```diff
@@ -285,6 +287,76 @@ static GOptionEntry deprecated_entries[] = {
     { NULL }
 };
 
+static void
+controller_event_cb(pcmk_ipc_api_t *controld_api,
+                    enum pcmk_ipc_event event_type, crm_exit_t status,
+                    void *event_data, void *user_data)
+{
+    pcmk_controld_api_reply_t *reply = event_data;
+
+    if (event_type != pcmk_ipc_event_reply) {
+        return;
+    }
+
+    if (status != CRM_EX_OK) {
+        exit_code = status;
+        g_set_error(&error, PCMK__EXITC_ERROR, exit_code,
+                    "Bad reply from controller: %s", crm_exit_str(exit_code));
+        return;
+    }
+
+    if (reply->reply_type != pcmk_controld_reply_info) {
+        exit_code = CRM_EX_PROTOCOL;
+        g_set_error(&error, PCMK__EXITC_ERROR, exit_code,
+                    "Unknown reply type %d from controller", reply->reply_type);
+        return;
+    }
+
+    if (reply->data.node_info.uname == NULL) {
+        exit_code = CRM_EX_NOHOST;
+        g_set_error(&error, PCMK__EXITC_ERROR, exit_code,
+                    "Node is not known to cluster");
+    }
+
+    exit_code = CRM_EX_OK;
+    pcmk__str_update(&options.dest_uname, reply->data.node_info.uname);
+}
+
+static int
+get_node_name_from_controller()
```

Comment:
needs (void) for some compilers (thank you debian-unstable)

Reply:
Fixed.